File: GNSDK/wrappers/gnsdk_csharp/src_wrapper/GnString.cs

namespace GracenoteSDK {

using System;
using System.Runtime.InteropServices;

/**
* Managed immutable string as returned by GNSDK.
*/
public class GnString : GnObject {
  private HandleRef swigCPtr;

  internal GnString(IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnString_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new HandleRef(this, cPtr);
  }

  internal static HandleRef getCPtr(GnString obj) {
    return (obj == null) ? new HandleRef(null, IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnString() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnString(swigCPtr);
        }
        swigCPtr = new HandleRef(null, IntPtr.Zero);
      }
      GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

		public override string
		ToString( )
		{
			string str = c_str( );
			return str;
		}
	
/**
* Construct an empty GnString object
*/
  public GnString() : this(gnsdk_csharp_marshalPINVOKE.new_GnString__SWIG_0(), true) {
  }

/**
* Construct a GnString object from a native constant string
* @param str [in] Native string
*/
  public GnString(string str) : this(gnsdk_csharp_marshalPINVOKE.new_GnString__SWIG_1(str), true) {
  }

/**
* Construct a GnString object from an existing GnString object
* @param str [in] GnString object
*/
  public GnString(GnString str) : this(gnsdk_csharp_marshalPINVOKE.new_GnString__SWIG_2(GnString.getCPtr(str)), true) {
    if (gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Pending) throw gnsdk_csharp_marshalPINVOKE.SWIGPendingException.Retrieve();
  }

  public string c_str() {
    string ret = gnsdk_csharp_marshalPINVOKE.GnString_c_str(swigCPtr);
    return ret;
  }

/**
* Get flag indicating if string object contains no string
* @return True of empty, false otherwise
*/
  public bool IsEmpty() {
    bool ret = gnsdk_csharp_marshalPINVOKE.GnString_IsEmpty(swigCPtr);
    return ret;
  }

/**
* Internally used factory for special SDK-managed strings
* @param str	[in] Native string
* @return Managed string
*/
  public static GnString manage(string str) {
    GnString ret = new GnString(gnsdk_csharp_marshalPINVOKE.GnString_manage(str), true);
    return ret;
  }

}

}

File: GNSDK/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnString.java

package com.gracenote.gnsdk;

/** 
* Managed immutable string as returned by GNSDK. 
*/ 
 
public class GnString extends GnObject {
  private long swigCPtr;

  protected GnString(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnString_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnString obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnString(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

		public String
		toString( )
		{
			String str = cStr( );
			return str;
		}
	
/** 
* Construct an empty {@link GnString} object 
*/ 
 
  public GnString() {
    this(gnsdk_javaJNI.new_GnString__SWIG_0(), true);
  }

/** 
* Construct a {@link GnString} object from a native constant string 
* @param str [in] Native string 
*/ 
 
  public GnString(String str) {
    this(gnsdk_javaJNI.new_GnString__SWIG_1(str), true);
  }

/** 
* Construct a {@link GnString} object from an existing {@link GnString} object 
* @param str [in] {@link GnString} object 
*/ 
 
  public GnString(GnString str) {
    this(gnsdk_javaJNI.new_GnString__SWIG_2(GnString.getCPtr(str), str), true);
  }

  public GnString set(GnString str) {
    return new GnString(gnsdk_javaJNI.GnString_set__SWIG_0(swigCPtr, this, GnString.getCPtr(str), str), false);
  }

  public GnString set(String str) {
    return new GnString(gnsdk_javaJNI.GnString_set__SWIG_1(swigCPtr, this, str), false);
  }

  public String cStr() {
    return gnsdk_javaJNI.GnString_cStr(swigCPtr, this);
  }

/** 
* Get flag indicating if string object contains no string 
* @return True of empty, false otherwise 
*/ 
 
  public boolean isEmpty() {
    return gnsdk_javaJNI.GnString_isEmpty(swigCPtr, this);
  }

/** 
* Internally used factory for special SDK-managed strings 
* @param str	[in] Native string 
* @return Managed string 
*/ 
 
  public static GnString manage(String str) {
    return new GnString(gnsdk_javaJNI.GnString_manage(str), true);
  }

}
